api/src/main/java/org/openmrs/module/ugandaemrreports/reports2019/SetupTxRTT2019Report.java

```diff
@@ -0,0 +1,192 @@
+        addIndicator(dsd,"PWIDSm","PWIDs TX Curr on ART male",df.getPatientsInAll(males,PWIDS,RTT),"");
+
+        addIndicator(dsd,"PIPf","PIPs TX Curr on ART female",df.getPatientsInAll(females,PIPS,RTT),"");
+        addIndicator(dsd,"PIPm","PIPs TX Curr on ART male",df.getPatientsInAll(males,PIPS,RTT),"");
+
+        return rd;
+    }
+
+    public void addGender(CohortIndicatorDataSetDefinition dsd, String key, String label, CohortDefinition cohortDefinition) {
+        if (key == "a") {
+            addIndicator(dsd, "2" + key, label, cohortDefinition, "age=below1female");
+            addIndicator(dsd, "3" + key, label, cohortDefinition, "age=between1and4female");
+            addIndicator(dsd, "4" + key, label, cohortDefinition, "age=between5and9female");
+            addIndicator(dsd, "5" + key, label, cohortDefinition, "age=between10and14female");
+            addIndicator(dsd, "6" + key, label, cohortDefinition, "age=between15and19female");
+            addIndicator(dsd, "7" + key, label, cohortDefinition, "age=between20and24female");
+            addIndicator(dsd, "8" + key, label, cohortDefinition, "age=between25and29female");
+            addIndicator(dsd, "9" + key, label, cohortDefinition, "age=between30and34female");
+            addIndicator(dsd, "10" + key, label, cohortDefinition, "age=between35and39female");
+            addIndicator(dsd, "11" + key, label, cohortDefinition, "age=between40and44female");
+            addIndicator(dsd, "12" + key, label, cohortDefinition, "age=between45and49female");
+            addIndicator(dsd, "13" + key, label, cohortDefinition, "age=above50female");
+        } else if (key == "b") {
+            addIndicator(dsd, "2" + key, label, cohortDefinition, "age=below1male");
+            addIndicator(dsd, "3" + key, label, cohortDefinition, "age=between1and4male");
+            addIndicator(dsd, "4" + key, label, cohortDefinition, "age=between5and9male");
+            addIndicator(dsd, "5" + key, label, cohortDefinition, "age=between10and14male");
+            addIndicator(dsd, "6" + key, label, cohortDefinition, "age=between15and19male");
+            addIndicator(dsd, "7" + key, label, cohortDefinition, "age=between20and24male");
+            addIndicator(dsd, "8" + key, label, cohortDefinition, "age=between25and29male");
+            addIndicator(dsd, "9" + key, label, cohortDefinition, "age=between30and34male");
+            addIndicator(dsd, "10" + key, label, cohortDefinition, "age=between35and39male");
+            addIndicator(dsd, "11" + key, label, cohortDefinition, "age=between40and44male");
+            addIndicator(dsd, "12" + key, label, cohortDefinition, "age=between45and49male");
+            addIndicator(dsd, "13" + key, label, cohortDefinition, "age=above50male");
+        }
+    }
+
+
+    public void addIndicator(CohortIndicatorDataSetDefinition dsd, String key, String label, CohortDefinition cohortDefinition, String dimensionOptions) {
```

Comment:
@dbaluku also is there a possibility to put this method in a publicly shared file such that it is usable accross different reports. we have many reports using this method and we define it in every report.

Reply:
Done